src/requests/completions.jl

```diff
@@ -10,8 +10,7 @@ end
 JSONRPC.parse_params(::Type{Val{Symbol("textDocument/completion")}}, params) = CompletionParams(params)
 function process(r::JSONRPC.Request{Val{Symbol("textDocument/completion")},CompletionParams}, server)
     if !haskey(server.documents, URI2(r.params.textDocument.uri))
-        send(JSONRPC.Response(r.id, CancelParams(r.id)), server)
-        return
+        error("Received 'textDocument/completion for non-existing document.")
```

Comment:
@ZacLN I'm slightly confused by these types of code block, for two reasons: 1) why would we ever be in this state? I would have expected that the extension would always first send a message that a file is opened, before it would ever send a message like this? So I guess it looks to me as if this is more a generic error state that we should never be in, and we should crash, send a report and figure out what is going on? 2) I think the whole `CancelParams` thing is not meant to be used like this? Or at least I can't find the part in the protocol that corresponds to this type of use. I thought `CancelParams` is only used in `$/cancelRequest` notifications, but those are sent not as a response to a request, but from the same party that started a request? I guess I just generally don't understand this part here :)

Reply:
1) Not sure how the server gets to that state, but in the past it did somehow. May no longer be needed as I think, in general, things are working a bit better - could test removal in the alpha.
2) Yep looks like an error